### PR TITLE
Fix phpunit deprecations

### PR DESCRIPTION
**Pull request by williamdes**
They should be static since a long time, phpunit is now unhappy if not static

Question: should we bump phpunit and migrate to annotations ?

### CODE REVIEW

**Comment by williamdes** [comment]
I pushed one last commit to address newer phpunit deprecations.

```
4 tests triggered 8 PHPUnit deprecations:

1) Slim\Tests\Psr7\ResponseTest::testWithStatusValidReasonPhraseObject
* MockBuilder::addMethods() is deprecated and will be removed in PHPUnit 12 without replacement.

* returnValue() is deprecated and will be removed in PHPUnit 12. Use $double->willReturn() instead of $double->will($this->returnValue())

/mnt/Dev/@debian/@wdes-packaging-team/php-slim-psr7/tests/ResponseTest.php:142

2) Slim\Tests\Psr7\UriTest::testWithHostValidObject
* MockBuilder::addMethods() is deprecated and will be removed in PHPUnit 12 without replacement.

* returnValue() is deprecated and will be removed in PHPUnit 12. Use $double->willReturn() instead of $double->will($this->returnValue())

/mnt/Dev/@debian/@wdes-packaging-team/php-slim-psr7/tests/UriTest.php:134

3) Slim\Tests\Psr7\UriTest::testWithQueryValidObject
* MockBuilder::addMethods() is deprecated and will be removed in PHPUnit 12 without replacement.

* returnValue() is deprecated and will be removed in PHPUnit 12. Use $double->willReturn() instead of $double->will($this->returnValue())

/mnt/Dev/@debian/@wdes-packaging-team/php-slim-psr7/tests/UriTest.php:299

4) Slim\Tests\Psr7\UriTest::testWithFragmentValidObject
* MockBuilder::addMethods() is deprecated and will be removed in PHPUnit 12 without replacement.

* returnValue() is deprecated and will be removed in PHPUnit 12. Use $double->willReturn() instead of $double->will($this->returnValue())

/mnt/Dev/@debian/@wdes-packaging-team/php-slim-psr7/tests/UriTest.php:351

```

**Comment by williamdes** [comment]
hello @akrabat 
what do you think about this PR and #310 ?

File: tests/TestObject.php [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ * Slim Framework (https://slimframework.com)
+ *
+ * @license https://github.com/slimphp/Slim-Psr7/blob/master/LICENSE.md (MIT License)
+ */
+
+declare(strict_types=1);
+
+namespace Slim\Tests\Psr7;
+
+final class TestObject implements \Stringable
```
Comment (akrabat) [inline tests/TestObject.php:13]: Can we rename to `StringableTestObject` or something along those lines? I'm not precious on the name, but something a little less generic would be helpful.

**Review by akrabat — CHANGES_REQUESTED** [review]

File: tests/TestObject.php (hunk 1 above)
Comment (williamdes) [reply tests/TestObject.php]: Good idea, I pushed a fix

**Comment by williamdes** [comment]
I also did a bump for the workflows and removed the need to require a library to upload coverage, this will help when the code will get incompatible with the uploader module. Now you are sure to always be able to upload coverage.
See: https://github.com/coverallsapp/github-action

**Review by akrabat — APPROVED** [review]

**Comment by akrabat** [comment]
Thanks.

**Pull request merged**